content/03-reference/01-tools-and-interfaces/01-prisma-schema/03-generators.mdx

```diff
@@ -4,158 +4,50 @@ metaTitle: 'Generators (Reference)'
-  binaryTargets = ["native"]
+  binaryTargets = ["linux-msl"]
 }
 ```
 
-In that case, Prisma detects your operating system and finds the right binary file for it based on the table [above](#supported-operating-systems). As you're running Mac OS, the binary file that was compiled for `darwin` will be selected.
-
-## Examples
+</TopBlock>
 
-**Specify the `prisma-client-js` generator with the default `output` and `binaryTargets`**
+## Binary targets
 
-```prisma
-generator client {
-  provider = "prisma-client-js"
-}
-```
+Prisma uses several [engine binaries](https://github.com/prisma/prisma-engines). For example:
 
-Note that the above `generator` definition is **equivalent** to the following because it uses the default values for `output` and `binaryTargets`:
+* **Prisma Client** uses the [query engine binary file](../prisma-client/query-engine) under the hood to communicate with a database. 
+* **Prisma Migrate** uses the migration engine binary file to run schema migrations against a database.
 
-```prisma
-generator client {
-  provider      = "prisma-client-js"
-  output        = "node_modules/@prisma/client"
-  binaryTargets = ["native"]
-}
-```
+Engines are implemented in Rust and are used by Prisma in the form of executable _binary files_.
 
-**Specify a custom `output` location for Prisma Client**
+The binary file is particularly important when [deploying](../prisma-client/deployment) your application to production.
 
-```prisma
-generator client {
-  provider = "prisma-client-js"
-  output   = "../src/generated/client"
-}
-```
+### The native binary target
 
-**Specify custom `binaryTargets` to ensure compatibility with the OS**
+The `native` binary target is special. It doesn't map to a concrete operating system. Instead, when `native` is specified in `binaryTargets`, Prisma detects the _current_ operating system and automatically specifies the correct binary target for it.
 
-This example shows how to configure Prisma Client to run on `Ubuntu 19.04 (disco)` based on the table [above](#linux-ubuntu).
+As an example, assume you're running **Mac OS** and you specify the following generator:
 
 ```prisma
 generator client {
   provider      = "prisma-client-js"
-  binaryTargets = ["debian-openssl-1.1.x"]
+  binaryTargets = ["native"]
 }
 ```
 
-**Specify a `provider` pointing to some custom generator implementation**
+In that case, Prisma detects your operating system and finds the right binary file for it based on the table [above](generators#supported-operating-systems). As you're running Mac OS, the binary file that was compiled for `darwin` will be selected.
 
-This example shows how to use a custom generator that's located in a directory called `my-generator`.
+### Custom binary targets
 
-```prisma
-generator client {
-  provider      = "./my-generator"
-}
-```
+You can use environment variables to [specify a custom binary](../../../understand-prisma/under-the-hood#using-custom-engine-binaries) for one or more engines.
```

Comment:
Technically the complete "Binary targets" section is documentation of Prisma Client JS, which let's you influence those via a configuration property in the `generator` block.

Reply:
Updated to specify `prisma-client-js` next to `binaries`